catalogmanagementv1/catalog_management_v1_examples_test.go

```diff
@@ -0,0 +1,1728 @@
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//
+// This file provides an example of how to use the Catalog Management service.
+//
+// The following configuration properties are assumed to be defined:
+// CATALOG_MANAGEMENT_URL=<service base url>
+// CATALOG_MANAGEMENT_AUTH_TYPE=iam
+// CATALOG_MANAGEMENT_APIKEY=<IAM apikey>
+// CATALOG_MANAGEMENT_AUTH_URL=<IAM token service base URL - omit this if using the production environment>
+//
+// These configuration properties can be exported as environment variables, or stored
+// in a configuration file and then:
+// export IBM_CREDENTIALS_FILE=<name of configuration file>
+//
+const externalConfigFile = "../catalog_mgmt.env"
+
+var (
+	catalogManagementService  *catalogmanagementv1.CatalogManagementV1
+	config                    map[string]string
+	configLoaded              bool = false
+	accountID                 string
+	bearerToken               string
+	gitAuthTokenForPublicRepo string
+	catalogID                 string
+	offeringID                string
+	clusterID                 string
+	objectID                  string
+	offeringInstanceID        string
+	versionLocatorID          string
+)
+
+func shouldSkipTest() {
+	if !configLoaded {
+		Skip("External configuration is not available, skipping tests...")
+	}
+}
+
+var _ = Describe(`CatalogManagementV1 Examples Tests`, func() {
+	Describe(`External configuration`, func() {
+		It("Successfully load the configuration", func() {
+			var err error
+			_, err = os.Stat(externalConfigFile)
+			if err != nil {
+				Skip("External configuration file not found, skipping tests: " + err.Error())
+			}
+
+			os.Setenv("IBM_CREDENTIALS_FILE", externalConfigFile)
+			config, err = core.GetServiceProperties(catalogmanagementv1.DefaultServiceName)
+			if err != nil {
+				Skip("Error loading service properties, skipping tests: " + err.Error())
+			}
+
+			configLoaded = len(config) > 0
+
+			accountID = config["ACCOUNT_ID"]
+			Expect(accountID).NotTo(BeEmpty())
+
+			clusterID = config["CLUSTER_ID"]
+			Expect(clusterID).NotTo(BeEmpty())
+
+			gitAuthTokenForPublicRepo = config["GIT_TOKEN"]
+			Expect(gitAuthTokenForPublicRepo).NotTo(BeEmpty())
+		})
+	})
+
+	Describe(`Client initialization`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+		It("Successfully construct the service client instance", func() {
+			var err error
+
+			// begin-common
+
+			catalogManagementServiceOptions := &catalogmanagementv1.CatalogManagementV1Options{}
+
+			catalogManagementService, err = catalogmanagementv1.NewCatalogManagementV1UsingExternalConfig(catalogManagementServiceOptions)
+
+			if err != nil {
+				panic(err)
+			}
+
+			// end-common
+
+			Expect(catalogManagementService).ToNot(BeNil())
+
+			token, err := catalogManagementService.Service.Options.Authenticator.(*core.IamAuthenticator).RequestToken()
+			Expect(err).To(BeNil())
+			bearerToken = token.RefreshToken
+			Expect(bearerToken).NotTo(BeNil())
```

Comment:
Just an info here. There is a new Go core release with the `GetToken()` method (it's exported now). The go.mod has been already updated on the main branch so if you merge in that, you could rewrite this to:
```suggestion
			bearerToken, err := catalogManagementService.Service.Options.Authenticator.(*core.IamAuthenticator).GetToken()
			Expect(err).To(BeNil())
			Expect(bearerToken).NotTo(BeNil())
```

Reply:
Thanks for the info! :)